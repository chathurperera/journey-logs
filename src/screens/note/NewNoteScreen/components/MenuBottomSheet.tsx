import React from 'react';
import { Text, View } from 'react-native';

export function MenuBottomSheet() {
  return (
    <View>
      <Text>MenuBottomSheet</Text>
    </View>
  );
}
