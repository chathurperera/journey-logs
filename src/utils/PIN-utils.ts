import moment from 'moment';

export const isPinSessionExpired = (lastSessionTimestamp: number, maxDurationInMinutes: number = 5) => {
  const timeDifference = moment().diff(moment(lastSessionTimestamp * 1000), 'minute');
  console.log('timeDifference', timeDifference);

  return timeDifference > maxDurationInMinutes;
};
