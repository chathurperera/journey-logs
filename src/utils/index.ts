export * from './moment-utils';
export * from './render-utils';
export * from './PIN-utils';
