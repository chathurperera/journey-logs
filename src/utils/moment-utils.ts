import moment from 'moment';

export const getCurrentTimestampInMilliSeconds = () => {
  const timestamp = moment().unix();
  return timestamp;
};

export const getRelativeTimeFromTimestamp = (timestampValue: number) => {
  const relativeTime = moment.unix(timestampValue).fromNow();
  return relativeTime;
};
