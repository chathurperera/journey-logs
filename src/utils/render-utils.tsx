import { NavigationContext } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React, { ReactNode } from 'react';
import { Provider } from 'react-redux';

import { store } from '@jl/stores';

type RenderParams = Parameters<typeof render>;

export function renderWithProviders(ui: RenderParams[0], options?: RenderParams[1]) {
  const BaseProviders = ({ children }: { children: ReactNode }) => (
    <Provider store={store}>
      <NavigationContext.Provider
        //@ts-ignore
        value={{
          isFocused: () => true,
          addListener: jest.fn(() => jest.fn()),
        }}>
        {children}
      </NavigationContext.Provider>
    </Provider>
  );

  return render(ui, { wrapper: BaseProviders, ...options });
}
