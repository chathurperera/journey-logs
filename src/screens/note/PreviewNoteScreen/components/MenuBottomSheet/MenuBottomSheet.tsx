import { Icon } from '@rneui/base';
import React, { forwardRef, useImperativeHandle, useRef } from 'react';
import { Pressable } from 'react-native';
import { Modalize } from 'react-native-modalize';

import { Text } from '@jl/components';
import { tw } from '@jl/config';
import { Color, Route, TextVariant } from '@jl/constants';
import { NavigationService, NoteService } from '@jl/services';

interface MenuBottomSheetProps {
  noteId: string;
  isEncrypted: boolean;
}

export const MenuBottomSheet = forwardRef(function MenuBottomSheet(
  { noteId, isEncrypted }: MenuBottomSheetProps,
  ref,
) {
  const ModalizeRef = useRef<Modalize>(null);

  useImperativeHandle(
    ref,
    () => ({
      open: () => {
        ModalizeRef.current?.open();
      },
    }),
    [],
  );

  const handleNoteEncryption = async () => {
    await NoteService.noteEncryption(noteId, !isEncrypted);
    ModalizeRef.current?.close();
  };

  const handleNoteDeletion = async () => {
    await NoteService.deleteNote(noteId);
    ModalizeRef.current?.close();
  };

  return (
    <Modalize ref={ModalizeRef} adjustToContentHeight>
      <Pressable
        onPress={() => NavigationService.navigate(Route.EditNote)}
        style={tw`border-b-[${Color.Primary.Jl150}] p-4 flex-row items-center gap-2 border-b-2`}>
        <Icon type="feather" name="edit" size={25} />
        <Text variant={TextVariant.Body1SemiBold} color={Color.Neutral.JL500}>
          Edit note
        </Text>
      </Pressable>
      <Pressable
        onPress={() => console.log('handle tags create screen navigation')}
        style={tw`border-b-[${Color.Primary.Jl150}] p-4 flex-row items-center gap-2 border-b-2`}>
        <Icon type="feather" name="tag" size={25} />
        <Text variant={TextVariant.Body1SemiBold} color={Color.Neutral.JL500}>
          Add Tags
        </Text>
      </Pressable>
      <Pressable
        onPress={handleNoteEncryption}
        style={tw`border-b-[${Color.Primary.Jl150}] p-4 flex-row items-center gap-2 border-b-2`}>
        <Icon type="feather" name={isEncrypted ? 'unlock' : 'lock'} size={25} />
        <Text variant={TextVariant.Body1SemiBold} color={Color.Neutral.JL500}>
          {isEncrypted ? 'Unlock note' : 'Lock note'}
        </Text>
      </Pressable>
      <Pressable
        onPress={handleNoteDeletion}
        style={tw`border-b-[${Color.Primary.Jl150}] p-4 flex-row items-center gap-2`}>
        <Icon type="feather" name="trash-2" color={Color.Warning.JL100} size={25} />
        <Text variant={TextVariant.Body1SemiBold} color={Color.Warning.JL100}>
          Delete Note
        </Text>
      </Pressable>
    </Modalize>
  );
});
