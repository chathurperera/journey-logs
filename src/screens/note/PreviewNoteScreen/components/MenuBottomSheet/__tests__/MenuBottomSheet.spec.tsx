import { act, cleanup, waitFor } from '@testing-library/react-native';
import { createRef } from 'react';
import { Modalize } from 'react-native-modalize';

import { renderWithProviders } from '@jl/utils';

import { MenuBottomSheet } from '../MenuBottomSheet';

describe('<MenuBottomSheet />', () => {
  const testID = 'MenuBottomSheetTestID';

  const mockProps = {
    testID: testID,
    noteId: 'sampleId',
    body: 'sampleBody',
    title: 'sampleTitle',
    isEncrypted: false,
    isFavourite: false,
    toggleEditingMode: jest.fn(),
  };

  const ref = createRef<Modalize>();
  const getRenderedComponent = () => renderWithProviders(<MenuBottomSheet {...mockProps} ref={ref} />);

  afterEach(cleanup);
  it('should render MenuBottomSheet correctly', () => {
    const renderer = getRenderedComponent();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the MenuBottomSheet via testID', async () => {
    const { getByTestId } = getRenderedComponent();
    expect(() => getByTestId(testID)).toThrow('Unable to find an element with testID: MenuBottomSheetTestID');

    act(() => {
      ref.current?.open();
    });

    await waitFor(() => {
      expect(() => getByTestId(testID)).toBeDefined();
    });
  });
});
