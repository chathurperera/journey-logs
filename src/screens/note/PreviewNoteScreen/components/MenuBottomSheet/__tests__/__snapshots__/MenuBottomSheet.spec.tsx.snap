// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBottomSheet /> should render MenuBottomSheet correctly 1`] = `null`;
